src/Chapter3.hs

```diff
@@ -971,6 +1082,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Eq, Ord, Enum, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend weekday =
+  case weekday of
+    Saturday -> False
+    Sunday -> False
+    _ -> True
+
+nextDay :: Weekday -> Weekday
+nextDay weekday
+  | weekday == maxBound = minBound
```

Comment:
Nice usage of standard functions! This implementation will work even if you decide to change the `Weekday` data type to the Americal system :+1:

Reply:
Yes... I am from Brazil, where the week starts on Sunday, but I lived for a while in Germany and Russia, where the week starts on Monday :)
I had my share of dealing with dates :joy: